library/std/src/sys/windows/ext/process.rs

```diff
@@ -98,6 +98,21 @@ pub trait CommandExt {
     /// [1]: https://docs.microsoft.com/en-us/windows/win32/procthread/process-creation-flags
     #[stable(feature = "windows_process_extensions", since = "1.16.0")]
     fn creation_flags(&mut self, flags: u32) -> &mut process::Command;
+
+    /// An option to enable quotes for all args when composite command line
+    /// that passed to the newly create process.
+    /// If the newly created process is a msys2/cygwin executable, the arg
+    /// with and without quotes have different meanings.
+    /// For arg without quotes and contains wildcard(*,?), msys2/cygwin will
+    /// automatically expand it by searching files with the `wildcard arg`
+    /// relative to the current dir.
+    /// For arg with quotes, msys2/cygwin will parse it and passed to main(argc,argv) without change.
+    /// Refer to  https://github.com/msys2/MSYS2-packages/issues/2176
+    /// In msvcrt based program, the commandline with and without quotes have
+    /// the same effect.
+    /// Refer to https://msdn.microsoft.com/en-us/library/17w5ykft.aspx
+    #[unstable(feature = "windows_process_extensions_force_quotes", issue = "77728")]
```

Comment:
It's unlikely but possible that this could be a breaking change. If someone implemented `CommandExt` in a separate crate, then adding this new associated function will break that implementation. I know, it's really unlikely, but still.

This could be prevented by either 1) moving `force_quotes` to a new trait, or 2) providing a default implementation of `force_quotes` which did nothing, but returned `self`. I'm not sure that's possible, though, given how the function signature is defined.

Reply:
Does trait are possible to implement defaut implementation?

Reply:
@lygstate Traits can have default implements for methods, if that's what you mean.

Reply:
@lygstate For example, you can do this:

```
pub trait Foo {
  fn foo(&self);   // <-- has no default

  fn bar(&self) {
    println!("I am the default behavior");
  }
}

```

Reply:
It's impossible to implement default implementation
```
    #[unstable(feature = "windows_process_extensions_force_quotes", issue = "77728")]
    fn force_quotes(&mut self, enabled: bool) -> &mut process::Command {
    }
```

Reply:
Well, if there's no reasonable default implementation that satisfies the expected behavior, then it doesn't make sense to _have_ a default implementation.

Reply:
However in this case you can probably just return `self` as @sivadeilra suggested. Having an empty body is definitely not going to work though.

Reply:
@camelid I am sorry for that, return `self` are also not working and I've tried.
It's trait and not inherit from `process::Command`, so return `self` not working